core/lib/zksync_core/src/reorg_detector/mod.rs

```diff
@@ -38,21 +38,31 @@ enum HashMatchError {
         Using an earlier snapshot could help."
     )]
     EarliestL1BatchTruncated(L1BatchNumber),
-    #[error("Internal error")]
-    Internal(#[from] anyhow::Error),
+    #[error("reorg detected, restart the node to revert to the last correct L1 batch #{0}.")]
+    ReorgDetected(L1BatchNumber),
+    #[error("storage error: {0:#}")]
+    Storage(anyhow::Error),
```

Comment:
Dumb question: Are `Internal` variant and the corresponding `From<anyhow::Error>` conversion too generic / implicit for your taste?

Reply:
I wanted to clarify that all the "internal" errors of these functions are actually storage errors.

Reply:
reverted `Storage` to `Internal`